dev-infrastructure/templates/maestro.bicep

```diff
@@ -0,0 +1,149 @@
+param deployMaestroPostgres bool = true
+
+@description('The name of the Postgres server for Maestro')
+@maxLength(60)
+param maestroPostgresServerName string
+
+@description('The version of the Postgres server for Maestro')
+param maestroPostgresServerVersion string
+
+
+@description('The size of the Postgres server for Maestro')
+param maestroPostgresServerStorageSizeGB int
+
+@description('If true, make the Maestro Postgres instance private')
+param maestroPostgresPrivate bool = true
+
+@description('The name of the managed identity that will manage certificates in maestros keyvault.')
+param maestroKeyVaultCertOfficerMSIName string = '${maestroKeyVaultName}-cert-officer-msi'
+
+@description('Deploys a Maestro Consumer to the management cluster if set to true.')
+param deployMaestroConsumer bool
+
+
+//
+// M A E S T R O  R E G I O N A L
+//
+
+module maestroInfra '../modules/maestro/maestro-infra.bicep' = {
+  name: 'maestro-infra'
+  params: {
+    eventGridNamespaceName: maestroEventGridNamespacesName
+    location: location
+    maxClientSessionsPerAuthName: maestroEventGridMaxClientSessionsPerAuthName
+    maestroKeyVaultName: maestroKeyVaultName
+    kvCertOfficerManagedIdentityName: maestroKeyVaultCertOfficerMSIName
+  }
+}
+
+//
+//   M A E S T R O  S E R V E R
+//
+
+resource vnet 'Microsoft.Network/virtualNetworks@2023-11-01' existing= {
+  name: 'aks-net'
+  scope: resourceGroup()
+}
+
+resource aksNodeSubnet 'Microsoft.Network/virtualNetworks/subnets@2023-11-01' existing= {
+  parent: vnet
+  name: 'ClusterSubnet-001'
+}
+
+var svcWorkloadIdentities = items({
+  maestro_wi: {
+    uamiName: 'maestro-server'
+    namespace: 'maestro'
+    serviceAccountName: 'maestro'
+  }
+})
+
+resource svcUami 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing= [
+  for wi in svcWorkloadIdentities: {
+    name: wi.value.uamiName
+    scope: resourceGroup(svcResourceGroup)
+  }
+]
+
+module maestroServer '../modules/maestro/maestro-server.bicep' = {
+  name: 'maestro-server'
+  scope: resourceGroup(svcResourceGroup)
+  params: {
+    maestroInfraResourceGroup: regionalResourceGroup
+    maestroEventGridNamespaceName: maestroEventGridNamespacesName
+    maestroKeyVaultName: maestroKeyVaultName
+    maestroKeyVaultOfficerManagedIdentityName: maestroKeyVaultCertOfficerMSIName
+    maestroKeyVaultCertificateDomain: maestroCertDomain
+    deployPostgres: deployMaestroPostgres
+    postgresServerName: maestroPostgresServerName
+    postgresServerVersion: maestroPostgresServerVersion
+    postgresServerStorageSizeGB: maestroPostgresServerStorageSizeGB
+    privateEndpointSubnetId: aksNodeSubnet.id
+    privateEndpointVnetId: vnet.id
+    postgresServerPrivate: maestroPostgresPrivate
+    maestroServerManagedIdentityPrincipalId: svcUami[0].properties.principalId
+    maestroServerManagedIdentityName: 'maestro-server'
+    location: location
+  }
+}
+
+//
+//   M A E S T R O  C O N S U M E R
```

Comment:
Infra to server to consumer is a 1:1:N relationship - I don't know that we ever need to be deploying the server and the consumer(s) in one pass, and even if we do, there would be potentially many consumers for each server.

Reply:
i split the maestro template to 3 different components and added the MVP bicepparam files as well. thanks for the advice !